src/Symfony/Component/DependencyInjection/Compiler/ValidateEnvPlaceholdersPass.php

```diff
@@ -37,7 +37,7 @@ public function process(ContainerBuilder $container)
     {
         $this->extensionConfig = [];
 
-        if (!class_exists(BaseNode::class) || !$extensions = $container->getExtensions()) {
+        if (!$container->isAvailable('symfony/config', BaseNode::class) || !$extensions = $container->getExtensions()) {
```

Comment:
this one should be reverted IMO as it is not about enabling a feature that affect later usages but about validating the placeholders